crates/iroha_core/src/state.rs

```diff
@@ -1434,8 +1434,8 @@ impl<'state> StateBlock<'state> {
     }
 
     /// Execute all time triggers matching the given block.
-    pub(crate) fn execute_time_triggers(&mut self, block: &SignedBlock) {
-        let time_event = self.create_time_event(block);
+    pub(crate) fn execute_time_triggers(&mut self, block_header: BlockHeader) {
```

Comment:
It should be sufficient to hold a reference to the `BlockHeader`:

```suggestion
    pub(crate) fn execute_time_triggers(&mut self, block_header: &BlockHeader) {
```

Reply:
Yes, it would. I pass plain `BlockHeader` because it is `Copy`, and previously Clippy told me that it is preferable not to reference `Copy` types.

Changed to `&BlockHeader` and didn't face warnings.